### PR TITLE
Data Export - Destroy Service

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the `DataExports` `DestroyService`. The service will eventually be called from the `data_exports_controller` and handles destroying the `DataExport` model instance as well as immediately purging the zip folder attached to the `DataExport`, if it still exists.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. In `rails c`, create a `data_export` as described in the [create service PR](https://github.com/phac-nml/irida-next/pull/444)
2. Assign the above export to variable `data_export = user.data_exports.last`
3. Delete the export with `DataExports::DestroyService.new(data_export, user).execute`
4. Ensure the export was deleted with `data_export.deleted?`

Test permission

5. Repeat the above steps, but assign a different user in the `DestroyService`
6. Ensure a permissions error occurs.

7. Ensure tests pass locally.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/services/data_exports/destroy_service.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  class DestroyService < BaseService
+    def initialize(data_export, user = nil)
+      super(user, {})
+      @data_export = data_export
+    end
+
+    def execute
+      authorize! @data_export, to: :destroy?
+
+      @data_export.file.blank? || @data_export.file.purge
```
Comment (ericenns) [inline app/services/data_exports/destroy_service.rb:14]: **nitpick (blocking):** Can you remove `acts_as_paranoid` and `deleted_at` column from `DataExport` model as we don't want these to be soft deletable. Can you also remove this line and instead update `has_one_attached` to include `dependant: :purge_later` so we don't have to wait for the file to be purged before returning from here. https://api.rubyonrails.org/v7.1.3.2/classes/ActiveStorage/Attached/Model.html#method-i-has_one_attached

**Review by ericenns — CHANGES_REQUESTED** [review]
Looking good, but I realized that the DataExport model is setup to be soft deletable which we don't want and I missed earlier.

File: app/services/data_exports/destroy_service.rb (hunk 1 above)
Comment (ChrisHuynh333) [reply app/services/data_exports/destroy_service.rb]: Fixed in 578a177b52473dcc8f9226cfdd4933cc1b96745e

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**